test/test_backpropagation.py

```diff
@@ -879,35 +880,47 @@ def test_backpropagate_timeout(self):
         slices = slice_by_depth(qc, 1)
         obs = SparsePauliOp("IX")
 
+        timeout_triggered = False
+
+        class MyTimeout(Exception):
+            def __init__(self):
+                nonlocal timeout_triggered
+                timeout_triggered = True
+
+        monkeypatch.setattr("qiskit_addon_obp.backpropagation.TimeoutException", MyTimeout)
+
         on_win = sys.platform == "win32"
-        with self.subTest("Actual timeout"):
+        with subtests.test(msg="Actual timeout"):
             if on_win:
                 pytest.skip("Does not run on Windows")
             many_slices = 100_000 * slices
-            t1 = time.time()
+
             _, new_qc, _ = backpropagate(obs, many_slices, max_seconds=1)
-            t2 = time.time()
 
-            with self.subTest("Time should be less than 2 seconds"):
-                assert (t2 - t1) < 2
+            with subtests.test(msg="Time should be less than 2 seconds"):
```

Comment:
Shouldn't this say `Time should be greater than 1 second`?

Reply:
No, because we want the timeout of `max_seconds` to ensure that the code does not run for too long (backpropagating the entire set of slices provided here would take significantly longer since its 100'000 slices).